### ISSUE TITLE
CLI leaves a mess behind from time to time

### ISSUE THREAD

**Issue by JustArchi** [crowdin/crowdin-cli-2#45]
```
Untracked files:
  (use "git add <file>..." to include in what will be committed)

        bg.zip
        de.zip
        el.zip
```

I have no clue under what conditions those files are popping up, but it seems that even after successful CLI run, one can find temporary zip files with updated translation lying around not being deleted.

This is not easily reproducable bug, I couldn't reproduce it with already pre-built translations. CLI version 2.0.14, I've just updated to latest 2.0.15, but I didn't see any commit mentioning this problem.

Thank you in advance.

**Comment by JustArchi** [comment]
Reproduced it again with 2.0.15, coincidentally the same zip files (bg, de and el) were left, so it doesn't look as random as I initially thought. Thanks!

**Comment by ihor-popyk** [comment]
Hello @JustArchi ,
do you work on Windows OS?

**Comment by JustArchi** [comment]
Hey @ihor-popyk, yes I'm using Windows for that. I noticed this bug started happening the moment crowdin-cli stopped downloading one big file and instead changed into per-locale compilation & downloading. Under some circumstances leftover files are left and need to be deleted manually - it doesn't seem to affect updated translation files though, they're being updated properly. Maybe crowdin-cli tries to remove still being used file too early? Linux allows removing files immediately, so it's possible that this bug is not even possible to reproduce there.

**Comment by ihor-popyk** [comment]
@JustArchi 
Have an idea, that cli doesn't have permissions to remove that files on your PC. Can you check it?

**Comment by JustArchi** [comment]
I can but I don't think that's the case, I have around ~40 target languages and only 3 zips out of at least 20 that are being compiled and deployed are left, while all other ones are being removed successfully.

Maybe you can add small debug of catching I/O exceptions around deleting the file, and add some basic info about them to stdout/stderr (maybe even with stacktrace), so when issue happens I can simply copy/paste what CLI writes. Personally I believe it has something to do with Windows Defender scanning the file and making it impossible to remove right away (file in use), but that's only my theory.

Thanks for looking into this issue!

**Comment by ihor-popyk** [comment]
Hey @JustArchi ,
Did some manipulation with CLI. Can you try to test this CLI version, please? 
[https://drive.google.com/file/d/0B9m3jUOiOAGLU1J2NGtnemVqTzQ/view?usp=sharing](url)

**Comment by JustArchi** [comment]
![image](https://user-images.githubusercontent.com/1069029/26909655-629481fa-4c02-11e7-86b0-0732f69e42d6.png)

Not sure if that output helps, but definitely I can reproduce the issue.

(Of course I have no issues removing those files from the same user as the one who executes crowdin-cli)

**Comment by JustArchi** [comment]
It seems even more random now, previously basically 3 languages made issues (in no-debug version), now it's far more often. Maybe it could lead to some clues.

![image](https://user-images.githubusercontent.com/1069029/26909700-bbec5afc-4c02-11e7-8c04-837520e9bdc5.png)

**Comment by ihor-popyk** [comment]
Hey @JustArchi 
Finally fixed it, just try 2.0.16
#49

**Comment by JustArchi** [comment]
Thank you a lot @ihor-popyk - I couldn't reproduce the issue with 2.0.16! 🙂
